### PR TITLE
vp8dec: enable svc-t decoding

### PR DESCRIPTION
**Pull request by wudping**
Support three layers vp8 svc-t decoding.

Signed-off-by: wudping <dongpingx.wu@intel.com>

### CODE REVIEW

**Comment by wudping** [comment]
please help to review, thanks.

File: decoder/vaapidecoder_base.h [hunk 1]
```diff
@@ -85,6 +85,8 @@ class VaapiDecoderBase:public IVideoDecoder {
 
     //do not use this, we will remove this in near future
     virtual VADisplay getDisplayID();
+
+    virtual bool isSVCTFrame() { return true; }
```
Comment (xuguangxin) [inline decoder/vaapidecoder_base.h]: Why do we need add this to the base?

File: decoder/vaapidecoder_vp8.cpp [hunk 2]
```diff
@@ -564,4 +567,28 @@ YamiStatus VaapiDecoderVP8::decode(VideoDecodeBuffer* buffer)
     return status;
 }
 
+bool VaapiDecoderVP8::isSVCTFrame()
+{
+    switch (m_configBuffer.temporalLayer) {
+    case 0:
```
Comment (xuguangxin) [inline decoder/vaapidecoder_vp8.cpp]: Please comments about 0, why we need return true for 0 layer

File: decoder/vaapidecoder_vp8.cpp [hunk 3]
```diff
@@ -564,4 +567,28 @@ YamiStatus VaapiDecoderVP8::decode(VideoDecodeBuffer* buffer)
...
+            return true;
+        break;
+    case 2:
+        if ((m_frameHdr.key_frame == Vp8FrameHeader::KEYFRAME) ||
+            m_frameHdr.refresh_last || m_frameHdr.refresh_golden_frame)
+            return true;
+        break;
+    case 3:
+        return true;
+    default:
```
Comment (xuguangxin) [inline decoder/vaapidecoder_vp8.cpp]: Please return true for default, if you have 3 layers, user want 4 layers, you need decode all 3 for them

File: interface/VideoDecoderDefs.h [hunk 4]
```diff
@@ -57,6 +57,13 @@ typedef struct {
     VAProfile profile;
     uint32_t flag;
     uint32_t fourcc;
+    //xxxLayer - layer number, 0: decode all layers, >0: decode up to target layer.
```
Comment (xuguangxin) [inline interface/VideoDecoderDefs.h]: Please remove unsupport things. vp8 not support it, not mean all codecs not support it.

Comment (xuguangxin) [reply interface/VideoDecoderDefs.h]: decode up to the target layer.-> How many layers you want to decode.

**Review by xuguangxin — CHANGES_REQUESTED** [review]

File: decoder/vaapidecoder_base.h (hunk 1 above)
Comment (wudping) [reply decoder/vaapidecoder_base.h]: Because other codecs may support SVC-T.

**Comment by wudping** [comment]
updated, thanks.

File: decoder/vaapidecoder_base.h (hunk 1 above)
Comment (xuguangxin) [reply decoder/vaapidecoder_base.h]: Different decoder may handle SVC in different way, this not needed in base class

File: decoder/vaapidecoder_vp8.cpp [hunk 5]
```diff
@@ -564,4 +567,27 @@ YamiStatus VaapiDecoderVP8::decode(VideoDecodeBuffer* buffer)
     return status;
 }
 
+bool VaapiDecoderVP8::isSVCTFrame()
```
Comment (xuguangxin) [inline decoder/vaapidecoder_vp8.cpp]: you can change this to isTargetFrame

**Comment by wudping** [comment]
updated, thanks @xuguangxin .

File: decoder/vaapidecoder_base.h [hunk 6]
```diff
@@ -85,6 +85,8 @@ class VaapiDecoderBase:public IVideoDecoder {
 
     //do not use this, we will remove this in near future
     virtual VADisplay getDisplayID();
+
+    virtual bool targetTemporalFrame() { return true; }
```
Comment (xuguangxin) [inline decoder/vaapidecoder_base.h]: Do not put this in base, it's codec specific feature

**Review by xuguangxin — CHANGES_REQUESTED** [review]

**Comment by wudping** [comment]
Updated, thanks @xuguangxin .

**Review by xuguangxin — APPROVED** [review]

**Review by xuguangxin — APPROVED** [review]

**Pull request merged**